### PR TITLE
Add Input Stream Support

### PR DESCRIPTION
**Pull request by hasithaa**

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/data/xml/Native.java [hunk 1]
```diff
@@ -19,25 +26,81 @@
...
 
+    private static final String METHOD_NAME_NEXT = "next";
+    private static final String METHOD_NAME_CLOSE = "close";
+
     public static Object fromXmlWithType(BXml xml, BMap<BString, Object> map, BTypedesc typed) {
         return null;
     }
 
-    public static Object fromXmlStringWithType(Object xml, BMap<BString, Object> map, BTypedesc typed) {
+    public static Object fromXmlStringWithType(Environment env, Object xml, BMap<BString, Object> map,
```
Comment (prakanth97) [inline native/src/main/java/io/ballerina/stdlib/data/xml/Native.java:36]: We have to update signature of Ballerina external function as well. Right?

File: native/src/main/java/io/ballerina/stdlib/data/xml/Native.java [hunk 2]
```diff
@@ -19,25 +26,81 @@
...
+                                                                            resolveNextMethod(iteratorObj),
+                                                                            resolveCloseMethod(iteratorObj),
+                                                                            resultConsumer)) {
+
+                    Object result = XmlParser.parse(new InputStreamReader(byteBlockSteam), typed.getDescribingType());
+                    future.complete(result);
+                } catch (Exception e) {
+                    return DataUtils.getError("Error occurred while reading the stream: " + e.getMessage());
+                }
+                return DataUtils.getError("Error occurred while reading the stream");
```
Comment (prakanth97) [inline native/src/main/java/io/ballerina/stdlib/data/xml/Native.java]: Here we always returning the error. Right?

File: native/src/main/java/io/ballerina/stdlib/data/xml/Native.java (hunk 1 above)
Comment (prakanth97) [reply native/src/main/java/io/ballerina/stdlib/data/xml/Native.java:36]: Seems no need. Ignore this comment.

File: native/src/main/java/io/ballerina/stdlib/data/xml/Native.java (hunk 2 above)
Comment (hasithaa) [reply native/src/main/java/io/ballerina/stdlib/data/xml/Native.java]: Good catch it was a bug.

**Pull request merged**